docs/tutorial/appendix/source-JSON-YAML/index.md

```diff
@@ -0,0 +1,274 @@
+
+Create a new file inside the `pages` folder, for this example with will be named `jsononclient.js`, but whatever name you decide is fine.
+
+Inside that file add the following:
+
+```javascript
+import React, { Component } from "react"
+import axios from "axios"
+import Layout from "../components/layout"
+import SEO from "../components/seo"
+
+class ClientJSON extends Component {
+  state = {
+    jsonData: {},
+  }
+  async componentDidMount() {
+    const JSONRequest = await axios.get("./second.json")
+    this.setState({ jsonData: JSONRequest.data })
+  }
+  render() {
+    const { jsonData } = this.state
+    if (!jsonData.title) {
+      return <h1>fetching data! give it a moment</h1>
+    }
+    return (
+      <Layout>
+        <SEO title={jsonData.title} />
+        <div>
+          {jsonData.content.map(data => {
+            return <div>{data.item}</div>
+          })}
+        </div>
+      </Layout>
+    )
+  }
+}
+
+export default ClientJSON
+```
+
+What the above code is doing is perfoming a GET request, using [Axios](https://github.com/axios/axios), the url when resolved, will load the contents of that particular static asset and update the component state in order to be used.
+
+**Be advised** that this block of code and the one bellow will function while developing your site. Some changes are required before issuing a production build.
+
+### Using YAML as a content provider a page
+
+Finally the same approach can be used for a YAML file, the process is somewhat similar to the above, instead of the file being copied to the static folder being the .json one, now is the .yaml one.
+
+Create file inside the `pages` folder, for this example it will be named `ymlonclient.js`, but like above, the name is entirely to the reader.
+
+Inside the newly create file, add the following contents:
+
+```javascript
+import React, { Component } from "react"
+import Layout from "../components/layout"
+import SEO from "../components/seo"
+import axios from "axios"
+import yaml from "js-yaml"
+class ClientYAML extends Component {
+  state = {
+    yamlData: {},
+  }
+  async componentDidMount() {
+    const ymlfile = await axios.get("./second.yaml")
+    const data = yaml.safeLoad(ymlfile.data)
+    this.setState({ yamlData: data })
+  }
+  render() {
+    const { yamlData } = this.state
+    if (!yamlData.title) {
+      return <h1>fetching data! give it a moment</h1>
+    }
+    return (
+      <Layout>
+        <SEO title={yamlData.title} />
+        <div>
+          {yamlData.content.map(data => {
+            return <div>{data.item}</div>
+          })}
+        </div>
+      </Layout>
+    )
+  }
+}
+export default ClientYAML
+```
+
+What the following code will do, is in fact similar to the one used for example mentioned above. The only diference is the way we handle the loading of the contents originating from the file.
```

Comment:
I think this paragraph should go above the code example...? Also there is a typo in the word "difference". Here's a suggestion on rewording if it is moved above:

"What the following code does is in fact similar to the other YAML example above. The only difference is..."